optd-persistent/src/migrator/memo/m20241029_000001_cascades_group.rs

```diff
@@ -1,3 +1,70 @@
+//! An entity representing a group / equivalence class in the Cascades framework.
+//!
+//! Quoted from the Microsoft article _Extensible query optimizers in practice_:
+//!
+//! > In the memo, each class of equivalent expressions is called an equivalent class or a group,
+//! > and all equivalent expressions within the class are called group expressions or simply
+//! > expressions.
+//!
+//! A Cascades group is defined as a class of equivalent logical or physical expressions. The
+//! Cascades framework uses these groups as a way of storing the best query sub-plans for use in the
+//! dynamic programming search algorithm.
+//!
+//! For example, a Cascades group could be the set of expressions containing the logical expressions
+//! `Join(A, B)` and `Join(B, A)`, as well as the physical expressions `HashJoin(A, B)` and
+//! `NestedLoopJoin(B, A)`.
+//!
+//! # Columns
+//!
+//! Each group is assigned a monotonically-increasing (unique) ID. This ID will be important since
+//! there are many foreign key references from other tables to `cascades_group`.
+//!
+//! We additionally store a `latest_winner` foreign key reference to a physical expression. See
+//! the [section](#best-physical-plan-winner) below for more details.
+//!
+//! Finally, we store `in_progress` and `is_optimized` flags that are used for quickly determining
+//! the state of optimization for this group during the dynamic programming search.
+//!
+//! # Entity Relationships
+//!
+//! ### Child Expressions (Logical and Physical)
+//!
+//! To retrieve all of a `cascades_group`'s equivalent expressions, you must query the
+//! [`logical_expression`] or the [`physical_expression`] entities via their foreign keys to
+//! `cascades_group`. The relationship between [`logical_expression`] and `cascades_group` is
+//! many-to-one, and the exact same many-to-one relationship is held for [`physical_expression`] to
+//! `cascades_group`.
+//!
+//! ### Parent Expressions (Logical and Physical)
+//!
+//! Additionally, each logical or physical expression can have any number of `cascades_group`s as
```

Comment:
No, groups are defined as a set of equivalent expressions, and so we can call an expression belonging to a group a "child" of that group.

Additionally, expressions will have groups as their children because the operators themselves have other child operators. So `Join(Group1, Group2)` has children `Group1`, and `Group2`.

See this a few lines down: https://github.com/cmu-db/optd-experimental/blob/0b1522994b78fdc525dfa2ae4f04da139f5c5f9c/optd-persistent/src/migrator/memo/m20241029_000001_cascades_group.rs#L45